.github/workflows/integration-install.yml

```diff
@@ -14,6 +14,8 @@ jobs:
       - name: Run Install Script
         shell: bash
         run: ./scripts/install.sh
+        env:
```

Comment:
☕️💡 question please - why as env? When we can use it as “secret” wherever it is needed 💭☕️

Reply:
if I pass it as secret to install.sh script , GitHub doesn't pass it and it is empty or blank for some security reasons. The only way I found is via environment variable. if you know a way to pass this TOKEN please let me know. Thank you.

Reply:
We should really-think this, passing gh-token like this means it can be scrapped into gh logs and defeat the purpose of anonymity which gh-token as default functionally provide.

For example - you can read some of it here (hugely downvoted ones) and some recommendations.

https://github.com/actions/starter-workflows/issues/785

build up question to this will be why we want to pass the gh-token to another script is next? 

But key reason to comment here is to bring this to your attention. Happy to focus more and love forward whatever best solution we could find, thanks heaps

Reply:
@Tatsinnit thanks for the details, but this is automated token generated by github for GH actions use. This cannot be obtained or converted to base 64 like it is posted in the above links/issues. I think above one is secrets used by your/any application that needs to be base 64'ed and stored as repository secrets which can be used by GH actions. 

More about automated token authentication - https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-2-calling-the-rest-api

Please correct me If I am wrong. 

**Note: this automated token expires in 24 hours if the job fails or once the action is completed it expires.**

Reply:
Sure, and more detail here https://docs.github.com/en/actions/security-guides/automatic-token-authentication

☕️Cool, feel free to make the decision my key aspect to mention was - are you suggesting 24 hours is safe window for creds not to be misused ? 🙂

build up to that is to think this scenario and if we can better workaround this- 

what is the scenario here which need gh-token to be passed to install script? Can we keep it all in this workflow instead? 


thank you so much

Reply:
💡 fyi to readers - Hari and I will discuss this and loop back, please feel free to reach out whatever is the easy way for any further thoughts around this.

thank you so much

Reply:
This is the [best practice and recommended approach](https://docs.github.com/en/actions/security-guides/encrypted-secrets#example-using-bash) that I've seen used across every workflow with a requirement like this. This should be good as long as we have [minimum permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) set for the token. 

Is there any documentation or discussions saying this pattern is not safe?

Reply:
💡💭☕️Cool, please note key thing I am keen to point out here is 2 folds: (second point for scenario of use is something which might be easy but again seems like there are more than 1 way to do this hence just discussing an idea where instead of api latest release we could read changelog action)

* Wether we choose to use GitHub-token or any other secret - the **permission** hardening should be minimalistic read here and how the permission and why we should permission tighten it - here is how we could do that - https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token

I think the permissions for the GITHUB_TOKEN are initially set to the default setting for the enterprise, organization, or repository. 

* Secondly, (not security related) have we thought of getting latest release number another way - I am not sure why token is needed for the install script scenario? Previously it was user with curl but without auth based request. What I can see is it used for only reading the released version? (So the intermittent failure was due to auth failures?)
    * option-1 - have we thought about or something came to me is that we maintain changelog for this repo and I wonder if it is mainly just to get the latest release number will it be easy to just get using changelog action avoid curl and we just pass that Release_VAR to install script? 
    * Option-2 - will it be simpler to maintain latest releases env var at repo level, it will be one click release version update.

Either way worth to archive this got clarity, thank you for the ping back.

Reply:
@OliverMKing @Tatsinnit Thanks for this valuable insights and thoughts. I have moved the step to get draft cli release tag to the workflow file to avoid passing GITHUB_TOKEN. here is the passing run - https://github.com/Azure/draft/actions/runs/5445153173/jobs/9903985731

If this is okay will clean up and looking for approvals. Thank you.